Makefile

```diff
@@ -195,14 +195,7 @@ endif
 # download controller-gen if necessary
 controller-gen:
 ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1 ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
+	@go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
```

Comment:
Thanks for the fix, but there are a few issues to be aware of. Do not install it directly. Install it in a temporary directory as before. Direct installation may cause go dependency problems.

Reply:
Also, version v0.6.2 is recommended, which performed well in my environment. From 0.4.1 to 0.14.0, the version span is too large.

Reply:
Got it! I'll make some fixes to my code.

Reply:
@tangming1996  May I ask what your `go version` is? My version is `1.22.4` and it still reproduced error when using `controller-gen==0.6.2`.

Reply:
The Go version in my current environment is 1.21.5.